tests/garage/tf/core/test_gru.py

```diff
@@ -0,0 +1,390 @@
+from nose2.tools.params import params
+import numpy as np
+import tensorflow as tf
+
+from garage.tf.core.gru import gru
+from tests.fixtures import TfGraphTestCase
+from tests.helpers import recurrent_step_gru
+
+
+class TestGRU(TfGraphTestCase):
+    def setUp(self):
+        super().setUp()
+        self.batch_size = 2
+        self.hidden_dim = 2
+
+        self.step_hidden_var = tf.placeholder(
+            shape=(self.batch_size, self.hidden_dim),
+            name='initial_hidden',
+            dtype=tf.float32)
+
+        self.gru_cell = tf.keras.layers.GRUCell(
+            units=self.hidden_dim,
+            activation=tf.nn.tanh,
+            kernel_initializer=tf.constant_initializer(1),
+            recurrent_activation=tf.nn.sigmoid,
+            recurrent_initializer=tf.constant_initializer(1),
+            name='lstm_layer')
+
+    # yapf: disable
+    @params(
+        (1, 1, 1, 0, 0),
+        (1, 1, 3, 0, 0),
+        (1, 3, 1, 0, 0),
+        (3, 1, 1, 0, 0),
+        (3, 3, 1, 0, 0),
+        (3, 3, 3, 0, 0),
+        (1, 1, 1, 0.5, 0.5),
+        (1, 1, 3, 0.5, 0.5),
+        (1, 3, 1, 0.5, 0.5),
+        (3, 1, 1, 0.5, 0.5),
+        (3, 3, 1, 0.5, 0.5),
+        (3, 3, 3, 0.5, 0.5))
+    # yapf: enable
+    def test_output_shapes(self, time_step, input_dim, output_dim, hidden_init,
+                           cell_init):
+        obs_inputs = np.full((self.batch_size, time_step, input_dim), 1.)
+        obs_input = np.full((self.batch_size, input_dim), 1.)
+
+        input_var = tf.placeholder(
+            tf.float32, shape=(None, None, input_dim), name='input')
+        step_input_var = tf.placeholder(
+            tf.float32, shape=(None, input_dim), name='input')
```

Comment:
Why do the two placeholders have the same name?

Reply:
Will fix